src/pages/guides/[...slug].astro

```diff
@@ -0,0 +1,23 @@
+---
+import { getCollection } from 'astro:content';
+import Layout from '../../components/Layout.vue';
+import BaseLayout from '../../layouts/BaseLayout.astro';
+import customComponents from '../../utils/customComponents';
+
+export async function getStaticPaths() {
+  const guides = await getCollection('guides');
+  return guides.map(entry => ({
+    params: { slug: entry.slug },
+    props: { entry },
+  }));
+}
+
+const { entry } = Astro.props;
+const { Content, headings } = await entry.render();
+const img = entry.data.img || '/default-cover-image.png';
+---
+<BaseLayout title={entry.data.title} description={entry.data.description} img={img}>
+  <Layout client:load collectionName="Guides" headings={headings} title={entry.data.title} img={img}>
```

Comment:
The `client:load` directive loads and hydrates our Vue component JavaScript immediately on page load.

There are [other directives](https://docs.astro.build/en/core-concepts/framework-components/#hydrating-interactive-components) - as we become more experienced with Astro, we can use a combination of these + Astro components to optimise our site performance